### PR TITLE
Increase uniswap deadline to 1 hr

### LINKED ISSUE #3316: Transaction failure
Another `exchange-exec-fail` when doing a swap... This is on appers to be creating a proxy and doing a swap with 1m gas limit.  The client reported an out of gas error, but etherscan doesn't really agree.  A parity debug trace shows one with `Bad jump destination`, thought might be a side effect.

`0x96880e15274ee05e242adfa2615140a410da169a96960393b6ee9a99f9045391`

    { blockHash:
       '0x003f5b7e94dab991064df8b7a6174bf44ce8600de840c9da42a4795ef382ab10',
      blockNumber: 8514734,
      from: '0x84294BB21adc7D5204c8a367C8416b6422f2BCfd',
      gas: 1000000,
      gasPrice: '15000000000',
      hash:
       '0x96880e15274ee05e242adfa2615140a410da169a96960393b6ee9a99f9045391',
      input:
       '0xc0394af1000000000000000000000000b8b530afcfc283ee330f9fcaea26468c4007c0c3000000000000000000000000000000000000000000000000000000000000008000000000000000000000000084294bb21adc7d5204c8a367c8416b6422f2bcfd000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002a4036c878100000000000000000000000084294bb21adc7d5204c8a367c8416b6422f2bcfd000000000000000000000000819bb9964b6ebf52361f1ae42cf4831b921510f900000000000000000000000000000000000000000000000000000000000000e000000000000000000000000009cabec1ead1c0ba254b09efb3ee13841712be14000000000000000000000000000000000000000000000000000000000000022000000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a232603590000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000000000010451b182500000000000000000000000000000000000000000000000000000000000000c360005b638eb2a0af0942e7f3f7af6637531c8ef334bd93000c948c8094d754c3600000000000000000000000000000000000000000000000000000000001275000000000000000000000000007ad0fa0e2380a5e0208b25ac69216bd7ff206bf800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a2326035900000000000000000000000064967e8cb62b0cd1bbed27bee4f0a6a2e454f06a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000446b1d4db70000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000005d76134f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
      nonce: 25,
      r:
       '0xabb713ebdc5d16539d464db0a67fbe5acc66e21e95391e5ebdfee7ed057cf1d7',
      s:
       '0x2bb541420bd174d675d8e6055ebc2a0bb8b74023326b95425a7c75675d428d7',
      to: '0x117793CC0b19C01c531638a986923533f2F865E7',
      transactionIndex: 121,
      v: '0x25',
      value: '5813218636572986' }
    { blockHash:
       '0x003f5b7e94dab991064df8b7a6174bf44ce8600de840c9da42a4795ef382ab10',
      blockNumber: 8514734,
      contractAddress: null,
      cumulativeGasUsed: 6912363,
      from: '0x84294bb21adc7d5204c8a367c8416b6422f2bcfd',
      gasUsed: 965459,
      logs: [],
      logsBloom:
       '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
      status: false,
      to: '0x117793cc0b19c01c531638a986923533f2f865e7',
      transactionHash:
       '0x96880e15274ee05e242adfa2615140a410da169a96960393b6ee9a99f9045391',
      transactionIndex: 121 }
    ============
    0x96880e15274ee05e242adfa2615140a410da169a96960393b6ee9a99f9045391
    --> 1. createProxyWithSenderNonce
    { _mastercopy: '0xb8B530afcfc283EE330F9FcaEa26468c4007c0C3',
      initializer:
       '0x036c878100000000000000000000000084294bb21adc7d5204c8a367c8416b6422f2bcfd000000000000000000000000819bb9964b6ebf52361f1ae42cf4831b921510f900000000000000000000000000000000000000000000000000000000000000e000000000000000000000000009cabec1ead1c0ba254b09efb3ee13841712be14000000000000000000000000000000000000000000000000000000000000022000000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a232603590000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000000000010451b182500000000000000000000000000000000000000000000000000000000000000c360005b638eb2a0af0942e7f3f7af6637531c8ef334bd93000c948c8094d754c3600000000000000000000000000000000000000000000000000000000001275000000000000000000000000007ad0fa0e2380a5e0208b25ac69216bd7ff206bf800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a2326035900000000000000000000000064967e8cb62b0cd1bbed27bee4f0a6a2e454f06a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000446b1d4db70000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000005d76134f00000000000000000000000000000000000000000000000000000000',
      _owner: '0x84294BB21adc7D5204c8a367C8416b6422f2BCfd',
      saltNonce: '0' }
    --> 2. swapAndMakeOffer
    { _owner: '0x84294BB21adc7D5204c8a367C8416b6422f2BCfd',
      _marketplace: '0x819Bb9964B6eBF52361F1ae42CF4831B921510f9',
      _offer:
       '0x51b182500000000000000000000000000000000000000000000000000000000000000c360005b638eb2a0af0942e7f3f7af6637531c8ef334bd93000c948c8094d754c3600000000000000000000000000000000000000000000000000000000001275000000000000000000000000007ad0fa0e2380a5e0208b25ac69216bd7ff206bf800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a2326035900000000000000000000000064967e8cb62b0cd1bbed27bee4f0a6a2e454f06a',
      _exchange: '0x09cabEC1eAd1c0Ba254B09efb3EE13841712bE14',
      _swap:
       '0x6b1d4db70000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000005d76134f',
      _token: '0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359',
      _value: '1000000000000000000' }

CC @micahalcorn @DanielVF @tomlinton @franckc

### PR DESCRIPTION
**Pull request by franckc**
### Description:

To handle the case of the network begin really slow, increase the deadline set for executing the DAI conversion from 5min to 1hr.

Also always use the local time to calculate the deadline - I did not see any advantage to using the latest block timestamp...

This should fix #3316 

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by mikeshultz** [comment]
Might want to hold off on this and finish our convo in #3316.  The concerns over using local time are pretty valid, I think.

**Comment by franckc** [comment]
Integrated review feedback. PTAL

File: packages/graphql/src/mutations/uniswap/swapToToken.js [hunk 1]
```diff
@@ -3,13 +3,23 @@ import contracts from '../../contracts'
...
+  //  - The local clock could be skewed
+  //  - When running in test/dev environment blocks may not be mined for some time.
   const blockNumber = await contracts.web3.eth.getBlockNumber()
   const block = await contracts.web3.eth.getBlock(blockNumber)
-
-  // If we're running on a private blockchain that hasn't mined a block
-  // recently, use a more recent timestamp
-  const now = Math.round(+new Date() / 1000)
-  const deadline = (block.timestamp < now - 60 ? now : block.timestamp) + 300
+  const localTimestamp = Math.round(+new Date() / 1000)
```
Comment (mikeshultz) [inline packages/graphql/src/mutations/uniswap/swapToToken.js]: What does the `+` mean in this context?

Comment (franckc) [reply packages/graphql/src/mutations/uniswap/swapToToken.js]: Fancy JS notation. It it the unary operator. It converts the date to a number.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus

Comment (mikeshultz) [reply packages/graphql/src/mutations/uniswap/swapToToken.js]: That's pretty cool, thanks.  Will keep it in mind.

**Pull request merged**